### ISSUE TITLE
AmazonS3Exception shown on logs as INFO and does not propagate

### ISSUE THREAD

**Issue by synhershko** [prestodb/presto#11242]
We are running the following with Presto, on an EC2 instance with the proper IAM role set up:

```sql
CREATE TABLE IF NOT EXISTS hive.default.elb_logs_pq (
             ->   request_timestamp varchar,
             ->   elb_name varchar,
             ->   request_ip varchar,
             ->   request_port int,
             ->   backend_ip varchar,
             ->   backend_port int,
             ->   request_processing_time double,
             ->   backend_processing_time double,
             ->   client_response_time double,
             ->   elb_response_code varchar,
             ->   backend_response_code varchar,
             ->   received_bytes bigint,
             ->   sent_bytes bigint,
             ->   request_verb varchar,
             ->   url varchar,
             ->   protocol varchar,
             ->   user_agent varchar,
             ->   ssl_cipher varchar,
             ->   ssl_protocol varchar )
             -> WITH ( format='PARQUET', external_location = 's3a://athena-examples/elb/parquet/');
```

Presto responds with the following:

```
Query 20180810_074251_00012_fjiwa failed: localhost:9083: java.net.SocketTimeoutException: Read timed out
```
In the logs we see the following:

```
2018-08-10T07:51:33.596Z        INFO    query-scheduler-9       com.amazonaws.latency   ServiceName=[Amazon S3], AWSErrorCode=[404 Not Found], StatusCode=[404], ServiceEndpoint=[https://athena-examples.s3.amazonaws.com], Exception=[com.amazonaws.services.s3.model.AmazonS3Exception: Not Found (Service: Amazon S3; Status Code: 404; Error Code: 404 Not Found; Request ID: 9356E0D9C871A828; S3 Extended Request ID: 4/KjybxBuBgGy5iL3z6gdZ39kr0BS+A7SRO4mJLEKf4T2/SIEOrG2ag++jNur/eUTMzAqnj2FfU=), S3 Extended Request ID: 4/KjybxBuBgGy5iL3z6gdZ39kr0BS+A7SRO4mJLEKf4T2/SIEOrG2ag++jNur/eUTMzAqnj2FfU=], RequestType=[GetObjectMetadataRequest], AWSRequestID=[9356E0D9C871A828], HttpClientPoolPendingCount=0, RetryCapacityConsumed=0, HttpClientPoolAvailableCount=0, RequestCount=1, Exception=1, HttpClientPoolLeasedCount=0, ClientExecuteTime=[38.05], HttpClientSendRequestTime=[0.089], HttpRequestTime=[37.803], RequestSigningTime=[0.084], CredentialsRequestTime=[0.002, 0.0], HttpClientReceiveResponseTime=[28.199], 
2018-08-10T07:51:33.609Z        INFO    query-scheduler-9       com.amazonaws.latency   ServiceName=[Amazon S3], StatusCode=[200], ServiceEndpoint=[https://athena-examples.s3.amazonaws.com], RequestType=[ListObjectsRequest], AWSRequestID=[22A574599C2BEA99], HttpClientPoolPendingCount=0, RetryCapacityConsumed=0, HttpClientPoolAvailableCount=1, RequestCount=1, HttpClientPoolLeasedCount=0, ResponseProcessingTime=[1.327], ClientExecuteTime=[13.048], HttpClientSendRequestTime=[0.049], HttpRequestTime=[11.593], RequestSigningTime=[0.06], CredentialsRequestTime=[0.001, 0.0], HttpClientReceiveResponseTime=[11.467], 
```

There are 3 related issues here:

1. Presto doesn't propagate the inner exception to the user, and instead responds with a 200 and "Read timeout", which is very confusing.
2. In the logs, the actual error is reported as INFO (and not ERROR) so might be easy to miss (or filtered, in real production systems)
3. The actual error message could be a bit more informative, e.g. report the S3 key the operation was tried on.

**Comment by nezihyigitbasi** [comment]
It's actually not Presto who is logging those lines, it's the `com.amazonaws.util.AWSRequestMetricsFullSupport` class in AWS sdk core library. You can open an issue against the AWS sdk java repo to ask about 2 and 3 above: https://github.com/aws/aws-sdk-java

The query `20180810_074251_00012_fjiwa ` fails with a socket read time out eventually -- there may be multiple retries under the hood that fail with completely different reasons, they will all be logged, but only the latest exception will be thrown by the retry logic.

**Comment by electrum** [comment]
If we really want the exceptions for each attempt, `RetryDriver` could add them as separate suppressed exceptions.

**Comment by synhershko** [comment]
Thanks @nezihyigitbasi ! Since this was a setup error (missing permissions) I expected the Presto CLI (or any other consumer) to surface that error. Could there maybe a way to show a "stacktrace"-like of errors leading to the final failure? I believe that was @electrum 's `RetryDriver` comment?

**Comment by synhershko** [comment]
Looks good! thanks!

**Comment by nezihyigitbasi** [comment]
I merged #11392 and it will go out with 0.211. Thanks for reporting this @synhershko.
